### DISCUSSION TITLE
Enhance `useGlobalPendingState` to avoid flickering or lingering loading indicators

### DISCUSSION THREAD

**Discussion by heivo** [sergiodxa/remix-utils#295] [Ideas]
I use the `useGlobalPendingState` hook in combination with https://github.com/frysztak/use-debounced-loader that let's me compute a derived loading state with an `initialDelay` and `minimalTimeOn`. This helps avoiding flickering spinners that are only displayed for a very short time.

It would be nice if the `useGlobalXxxSate` hooks could provide this functionality directly as optional parameters, e.g. `const state = useGlobalPendingState({initialDelay: 150, minimalTimeOn: 250})`.

**Comment by sergiodxa** [comment]
I think you can combine it with spin-delay package to achieve this in a few lines of code:

```ts
let globalState = useGlobalPendingState()
let isPending = showSpinner = useSpinDelay(globalState !== "idle", {
  delay: 500, minDuration: 200
})
```

**Reply by heivo** [reply]
Seems like `useSpinDelay` and `useDebouncedLoader` do the same. But you are right, probably better to keep the functionality separated. Thank you.
